sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -12,18 +15,75 @@ namespace core
 namespace http
 {
 
-class http_request
+namespace http_method
+{
+enum HttpMethod
 {
+  GET,
+  HEAD,
+  POST,
+  PUT,
+  DELETE,
+  PATCH,
+};
+}
 
+class Header
+{
 private:
-  /* data */
+  std::string name;
+  std::string value;
+
 public:
-  http_request();
-  ~http_request();
+  Header(std::string name, std::string value)
+  {
+    this->name = name;
+    this->value = value;
+  }
+  ~Header() {}
+  std::string getName();
+  std::string getValue();
 };
 
-http_request::http_request() {}
-http_request::~http_request() {}
+class Request
+{
+
+private:
+  http_method::HttpMethod method;
+  std::string url;
+  std::vector<Header> headers;
+  std::string body;
+  size_t query_start; // 0 = no query in url, > 0 = query start position in url '?'
+
+  inline size_t get_query_start(std::string someUrl)
+  {
+    auto position = someUrl.find('?');
+    return position == std::string::npos ? 0 : position;
```

Comment:
By converting `std::string::npos` to `0`, are you suggesting `someUrl` cannot start with `?`?

Reply:
We are currently not checking if url is valid or not. But having a url starting with `?` is definitely invalid.
url validation would try to find schema first (http:://), then a host, and then optional sections.
We have some code for parsing url in C99 code, we might improve this in the future to check for valid url